packages/core/react/src/WalletProvider.tsx

```diff
@@ -98,6 +98,13 @@ export const WalletProvider: FC<WalletProviderProps> = ({
         }
     }, [name, enhancedWalletsByName, setState]);
 
+    // When the adapter changes, disconnect the old one.
+    useEffect(() => {
+        return () => {
+            adapter?.disconnect().catch((error) => console.warn(error));
+        };
+    }, [adapter]);
+
```

Comment:
If you move this above the hook that cleans up the adapter's event listeners, I _think_ that calling `disconnect()` on line 104 will _fire_ the disconnect event on line 164, which will have some unpredictable consequences.

Reply:
Best to let the adapter's event handler hook disconnect the listeners, and _then_ call `disconnect` on the adapter!

Reply:
lol yep I broke this 😅